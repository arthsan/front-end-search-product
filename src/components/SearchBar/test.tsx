import { screen, render } from '@testing-library/react'
import { QueryClientProvider } from 'react-query'
import { queryClient } from 'services/queryClient'

import { SearchBar } from '.'

describe('<SearchBar />', () => {
  it('should be able to click on button', () => {
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <SearchBar />
      </QueryClientProvider>
    )

    const buttonElement = screen.getByRole('fetch')

    expect(buttonElement).toBe(buttonElement)
    expect(buttonElement).toHaveTextContent('Search')
    expect(container.firstChild).toMatchSnapshot()
  })
})
