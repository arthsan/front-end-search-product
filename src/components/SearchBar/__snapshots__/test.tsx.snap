// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should be able to click on button 1`] = `
<div
  class="css-w4gwbl"
>
  <div
    class="css-1lekzkb"
  >
    <div
      class="chakra-input__group css-4302v8"
    >
      <div
        class="chakra-input__left-element css-kn13u6"
      >
        <svg
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <input
        class="chakra-input css-1q2zmi9"
        name="item"
        placeholder="Search for a product (camiseta)"
      />
    </div>
    <button
      class="chakra-button css-cpb45r"
      role="fetch"
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;
